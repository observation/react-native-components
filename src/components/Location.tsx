import React from 'react'
import { StyleProp, ViewStyle } from 'react-native'

import { Icon } from './Icon'
import IconText from './IconText'
import textStyle from '../styles/text'
import theme from '../styles/theme'

type Props = {
  location: string
  containerStyle?: StyleProp<ViewStyle>
}

const Location = ({ location, containerStyle }: Props) => (
  <IconText
    icon={<Icon name="map-marker-alt" style="solid" color={theme.color.greySemi} size={theme.icon.size.medium} />}
    text={location}
    style={{
      containerStyle,
      textStyle: textStyle.light,
    }}
    singleLineText
  />
)

export default Location
