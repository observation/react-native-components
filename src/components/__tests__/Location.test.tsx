import React from 'react'

import { render } from '@testing-library/react-native'

import Location from '../Location'

describe('Location', () => {
  test('Rendering', () => {
    const { toJSON } = render(
      <Location location="Voorschoten - algemeen (Zuid-Holland)" containerStyle={{ flex: 1 }} />,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
