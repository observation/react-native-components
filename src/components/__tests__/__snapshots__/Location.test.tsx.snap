// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location Rendering 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "justifyContent": "center",
          "marginRight": 8,
        },
        undefined,
      ]
    }
  >
    <Icon
      color="#E6E6E6"
      name="location-dot"
      size={14}
      type="solid"
    />
  </View>
  <Text
    numberOfLines={1}
    style={
      [
        {
          "flexShrink": 1,
        },
        {
          "color": "#939393",
          "fontFamily": "Ubuntu",
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "normal",
          "lineHeight": 20,
        },
      ]
    }
  >
    Voorschoten - algemeen (Zuid-Holland)
  </Text>
</View>
`;
